### ISSUE TITLE
JWT validation failure

### ISSUE THREAD

**Issue by dmakarenko-connyun** [ktorio/ktor#676]
The #664 was fixed, thanks. But a new issue was introduced in beta-2 in the following part:
```
        try {
            val principal = verifyAndValidate(call, verifier(token), token, schemes, authenticate)
            if (principal != null) {
                context.principal(principal)
            } else {
                context.bearerChallenge(AuthenticationFailedCause.InvalidCredentials, realm, schemes)
            }
        } catch (cause: Throwable) {
            val message = cause.message ?: cause.javaClass.simpleName
            JWTLogger.trace("JWT verification failed: {}", message)
            context.error(JWTAuthKey, AuthenticationFailedCause.Error(message))
        }
```
If an exception is thrown in the `validate` function which can be defined by user in configuration then the authentication is not failed and the server returns 200 OK
```
09:10:46.985 [DefaultDispatcher-worker-1 @request#3] TRACE io.ktor.auth.jwt [JWTAuth.kt:127] - JWT verification failed: Something bad happened in validate function.
09:10:47.024 [DefaultDispatcher-worker-1 @request#3] TRACE ktor.test [CallLogging.kt:138] - 200 OK: GET - /me
```

**Comment by cy6erGn0m** [comment]
Looks like authentication doesn't handle errors reported by providers at all

**Comment by cy6erGn0m** [comment]
Should be fixed in `1.0.0-beta-3`

**Comment by dmakarenko-connyun** [comment]
Works. Thanks.
